### ISSUE TITLE
devServer question/feature request

### ISSUE THREAD

**Issue by vegarringdal** [fuse-box/fuse-box#294]
@nchanged @devmondo 

Is it possible to start the devServer without it listening to anything?
Also is it possible to call a devServer to force it to reload ?

Been thinking.. (yes, for real:joy:) 

If I can start a devserver and attach emitter on it, then I can use the change event to trigger update on the dev server serving the pages.

That way I can work on many packages at once now, like a plugin and sample app that uses the plugin, yes Aurelia... only thing I use :-)

So if this is possible already then great, if not, is it something you guys would be willing to consider? if its easy to implement.

Anyway hope everything is going great with the BTI

**Comment by unlight** [comment]
> Also is it possible to call a devServer to force it to reload
I have written a plugin for that https://github.com/unlight/fusebox-reload
You need change config, see there

**Comment by vegarringdal** [comment]
Yes for the reloading part I can use that one

**Comment by nchanged** [comment]
@vegarringdal so what's the feature request here?;-)

**Comment by vegarringdal** [comment]
@nchanged 

If I can call/send socket event from nodeprocess running the devServer then I guess the feature request is being able to run devserver without it listening to any ports so I dont haveto worry about port confilicts.
That way I only use it to watch a bundle/folder/package


something like this maybe... (dunno if I can call the socket server like this, not at my pc, so cant test
```
var server1, server2
server1 =bundle1.devServer(">index.ts", {
    emitter: (self, fileInfo) => {
        self.socketServer.send("source-changed", fileInfo);
    }
});

server2 =bundle2.devServer(">index.ts", {
   onlyWatch: true,
    emitter: (self, fileInfo) => {
        server1.socketServer.send("source-changed", fileInfo)
    }
});

**Comment by Thanood** [comment]
Personally, my use-case for this feature would be a JS lib with a sample app in a subfolder.. [Like this one](https://github.com/aurelia-ui-toolkits/aurelia-materialize-bridge) (currently not using fuse-box, I'm working on it :smirk: ).

There is a `sample` folder which would be used with devServer during development of the library (`src` folder). Once the sample app is stable, the library will be the source which constantly changes.

In that case, i would want the devServer to refresh (or hot-reload) whenever I change the library source.

**Comment by unlight** [comment]
@vegarringdal
> That way I only use it to watch a bundle/folder/package

You can use standalone watch utils from nodeprocess like `fs.watchFile` or there are many npm packages which can do that: chokidar, watchman, gulp.watch
I'm using `gulp.watch`, you can see example [https://github.com/unlight/angular-fusebox-seed/blob/eebf2c0a38e2c518be17820c7b30681be7095a3b/gulpfile.ts#L220-L228](here)

**Comment by vegarringdal** [comment]
@unlight 
Yes I could just do that, but having a watch feature built in could be nice, I could maybe try and make a plugin, better team spend time on the bugs first so they get get to the BTI and features 👍

**Comment by devmondo** [comment]
@vegarringdal FuseBox do have watch feature as a second param to `FuseBox.bundle(glob..., watch:true)`. of course we need more :)


@nchanged  they will love it when we move to next phase!

**Comment by vegarringdal** [comment]
@devmondo 
nioce, I dont think there is anything about that part in the docs

**Comment by nchanged** [comment]
@devmondo no it has been removed ;-)

**Comment by vegarringdal** [comment]
ahh, you dont want people to use it ?

**Comment by vegarringdal** [comment]
or the code have been removed?

**Comment by devmondo** [comment]
@nchanged  crap! @vegarringdal  not all hope is gone, soon, soon.....

**Comment by devmondo** [comment]
solved in [milestone 1.3.123](https://github.com/fuse-box/fuse-box/milestone/11)

please reopen if not :)
